forum/Sources/Post.php

```diff
@@ -1188,12 +1188,16 @@ function Post()
 	$context['require_verification'] = !$user_info['is_mod'] && !$user_info['is_admin'] && !empty($modSettings['posts_require_captcha']) && ($user_info['posts'] < $modSettings['posts_require_captcha'] || ($user_info['is_guest'] && $modSettings['posts_require_captcha'] == -1));
 	if ($context['require_verification'])
 	{
-		require_once($sourcedir . '/Subs-Editor.php');
-		$verificationOptions = array(
-			'id' => 'post',
-		);
-		$context['require_verification'] = create_control_verification($verificationOptions);
-		$context['visual_verification_id'] = $verificationOptions['id'];
+		// Use SMF functionality if not using wetfish captcha.
+		if ($modSettings['visual_verification_type'] != 6)
+		{
+			require_once($sourcedir . '/Subs-Editor.php');
+			$verificationOptions = array(
+				'id' => 'post',
+			);
+			$context['require_verification'] = create_control_verification($verificationOptions);
```

Comment:
while your fix seems to work, i've tested it every way i could think of, i think maybe exploring create_control_verification would be a good idea, that might be the root cause of why that variable was null.